opm/core/wells/WellCollection.hpp

```diff
@@ -36,11 +36,11 @@ namespace Opm
     {
     public:
 
-        void addField(const Group& fieldGroup, size_t timeStep, const PhaseUsage& phaseUsage);
+        void addField(const Group2& fieldGroup, size_t timeStep, const PhaseUsage& phaseUsage);
 
         void addWell(const Well2& wellChild, const SummaryState& summaryState, size_t timeStep, const PhaseUsage& phaseUsage);
```

Comment:
Similar to the Group2, with Well2, the `timeStep` should not be needed here too, while it is used to give a warning message. 

In some way, with the new `Group2` and `Well2`, we already have the information related to group and well for the specified `report step`, and at the same time, we do not know the information about the `report step` through them.  I am not sure what is the best way to keep both information.